File: recursion/recursion-examples/visual-basic/nth_term.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-SA 4
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program


Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim n As Integer = 6 ' Test data
        Dim increment As Integer = 3 ' Test data
        Dim result As Integer = NthTerm(n, increment)
        Console.WriteLine($"Term {n} is {result}")
        Console.ReadLine()
    End Sub

    Function NthTerm(ByVal n As Integer, ByVal increment As Integer) As Integer
        If n = 1 Then ' Base case
            Return 1
        Else
            Return increment + NthTerm(n - 1, increment) ' General case
        End If
    End Function

End Module

File: recursion/recursion-examples/java/nth_term.java
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science,
part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

class RecursionExamples
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args)
    {
        int n = 6;
        int increment = 3;
        int result = nthTerm(n, increment);
        System.out.println("Term " + n + " is " + result);
    }


    // Returns the nth term of a numerical sequence
    public static int nthTerm(int n, int increment)
    {
        if (n == 1) {
            return 1;
        }
        else {
            return increment + nthTerm(n-1, increment);
        }
    }


}